### PR TITLE
A few added fields for ES tuning

### PR DESCRIPTION
**Pull request by nonword**
Adds a couple new ES fields to improve author matching.

Also fixes a couple minor things with bulk-index and compare-with-indexed scripts.

~This depends on Node20 updates, so targets that branch currently, but will be merged into `main` when which are now in `main` since [PR 89](https://github.com/NYPL/research-catalog-indexer/pull/89) is merged.~

### CODE REVIEW

File: lib/utils/author-names.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }
+  }
+
+  return permutations
+}
+
+const withoutDates = (name) => {
+  if (typeof name !== 'string') return name
+
+  const tokens = name.split(', ')
```
Comment (charmingduchess) [inline lib/utils/author-names.js:41]: I'm struggling to find an example, but I imagine there could be a case of a mononymous author with dates?

**Review by charmingduchess — APPROVED** [review]
Looks good, just pointed out an edge case that might need addressing.

File: lib/utils/author-names.js (hunk 1 above)
Comment (nonword) [reply lib/utils/author-names.js:45]: Cher!
```
"creatorLiteral": "Cher, 1946-"
```

Good catch. Fixed

Comment (charmingduchess) [reply lib/utils/author-names.js:45]: That's so funny. She was the only person I could think of without googling a list of authors and I never thought to try her.

**Review by charmingduchess — APPROVED** [review]

**Pull request merged**